qemu/tests/migration_with_numa.py

```diff
@@ -0,0 +1,111 @@
+import re
+import logging
+
+from avocado.utils import process
+from virttest import env_process
+from virttest import error_context
+from virttest.utils_misc import get_mem_info
+from virttest.utils_misc import normalize_data_size
+from virttest.utils_misc import NumaInfo
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Bind guest node0 and node1 to 2 host nodes, do migration test
+
+    1. Boot src guest with 2 numa node and all bind to 2 host numa nodes
+    2. Migration
+    3. Check the memory in guest
+    4. Check the memory in dest host
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    def get_nodes_size(size_type='MemTotal', session=None):
+        """
+        Get the node size of each node in host/guest, descending sort with size
+
+        :param size_type: the type of the node size
+        :param session: ShellSession object
+
+        :return: a list of tuple include node id and node size
+        :rtype: list
+        """
+        numa_info = NumaInfo(session=session)
+        nodes_size = {}
+        numa_nodes = numa_info.online_nodes
+        for node in numa_nodes:
+            nodes_size[node] = float(numa_info.online_nodes_meminfo[node][size_type])
+        nodes_size = sorted(nodes_size.items(), key=lambda item: item[1], reverse=True)
+        return nodes_size
+
+    host_nodes_size = get_nodes_size(size_type='MemFree')
+    mem_devs = params.objects('mem_devs')
+    for mem_dev in mem_devs:
+        size_mem = params.object_params(mem_dev).get('size_mem')
+        size_mem = float(normalize_data_size(size_mem, 'K'))
+        if host_nodes_size[0][1] >= int(size_mem):
+            params['host-nodes_mem_%s' % mem_dev] = str(host_nodes_size[0][0])
+            del host_nodes_size[0]
+        else:
+            test.cancel("host nodes do not have enough memory for testing!")
+
+    params['start_vm'] = 'yes'
+    env_process.preprocess(test, params, env)
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    # do migration
+    mig_timeout = float(params.get("mig_timeout", "3600"))
+    mig_protocol = params.get("migration_protocol", "tcp")
+    vm.migrate(mig_timeout, mig_protocol, env=env)
```

Comment:
Does src qemu quit after migration?

Reply:
> Does src qemu quit after migration?

Yes, `vm.migrate` do it, you can refer:
https://github.com/avocado-framework/avocado-vt/blob/b3403614070a543dac3f1a9d9e11f4e1a70e512c/virttest/qemu_vm.py#L4393